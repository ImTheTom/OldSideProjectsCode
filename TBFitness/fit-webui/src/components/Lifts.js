import React from "react";
import Header from "./Header";
import { connect } from "react-redux";
import { lifts } from "../actions/fit";
import DisplayLift from "./LiftsPage/DisplayLift";
import axios from "axios";
import Config from "../config";

class Lifts extends React.Component {
	render() {
		let display;
		if (Object.keys(this.props.liftsData).length !== 0) {
			display = Object.values(this.props.liftsData).map((lift, index) => (
				<DisplayLift lifts={lift} liftIndex={index} key={index}/>
			));
		}
		return (
			<div>
				<div>
					<Header />
				</div>
				<div className="lifts-box">
					<div className="create-lift-title">
					Lifts
					</div>
					{display}
				</div>
			</div>
		);
	}

	componentDidMount() {
		axios.get(Config.host.NAME + "lift/1").then(({ data }) => {
			this.props.lifts(data);
		});
	}
}

const mapStateToProps = state => ({
	liftsData: state.lifts
});

const mapDispatchToProps = (dispatch) => ({
	lifts: (data) => dispatch(lifts(data)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Lifts);

