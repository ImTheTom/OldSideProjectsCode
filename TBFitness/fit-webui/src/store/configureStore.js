import { createStore, combineReducers, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import workout from "../reducers/workout";
import lifts from "../reducers/lifts";
import editWorkout from "../reducers/editWorkout";

const store = createStore(combineReducers({ workout: workout, lifts: lifts, editWorkout: editWorkout }), applyMiddleware(thunk));

export default store;

