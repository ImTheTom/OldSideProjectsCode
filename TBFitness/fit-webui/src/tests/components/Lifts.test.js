import Lifts from "../../components/Lifts";
import React from "react";
import { shallow } from "enzyme";
import liftsBase from "../fixtures/liftsBase";
import store from "../../store/configureStore";
import { Provider } from "react-redux";

test("should render diplay lift component correctly", () => {
	const wrapper = shallow(
		<Provider store={store}>
			<Lifts liftsData={liftsBase}/>
		</Provider>
	);
	expect(wrapper).toMatchSnapshot();
});