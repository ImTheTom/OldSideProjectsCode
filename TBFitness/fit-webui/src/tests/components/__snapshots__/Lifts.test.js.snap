// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render diplay lift component correctly 1`] = `
<Connect(Lifts)
  liftsData={
    Object {
      "Lifts": Array [
        Object {
          "id": 1,
          "name": "BenchPress",
          "reps": 5,
          "sets": 5,
          "user_id": 1,
          "weight": 20,
        },
        Object {
          "id": 2,
          "name": "Overhead Press",
          "reps": 10,
          "sets": 3,
          "user_id": 1,
          "weight": 10,
        },
        Object {
          "id": 3,
          "name": "Incline Dumbell Press",
          "reps": 10,
          "sets": 3,
          "user_id": 1,
          "weight": 15,
        },
        Object {
          "id": 4,
          "name": "Tricep Pulldown",
          "reps": 10,
          "sets": 3,
          "user_id": 1,
          "weight": 10,
        },
      ],
    }
  }
/>
`;
